go/roothash/api/block/header.go

```diff
@@ -17,6 +19,16 @@ var ErrInvalidVersion = errors.New("roothash: invalid version")
 // HeaderType is the type of header.
 type HeaderType uint8
 
+// TimestampType is a custom time stamp type that encodes like time.Time when
+// marshaling to JSON.
+type TimestampType uint64
+
+// MarshalJSON encodes the time stamp to JSON by converting it from Unix time to
+// local time.
+func (tst TimestampType) MarshalJSON() ([]byte, error) {
```

Comment:
What about unmarshalling? Add a test that makes sure this round-trips correctly.

Reply:
Added unmarshaling and tests.